tiatoolbox/models/engine/io_config.py

```diff
@@ -0,0 +1,390 @@
+        """Returns a new config object converted to baseline form.
+
+        This will return a new :class:`IOSegmentorConfig` where
+        resolutions have been converted to baseline format with the
+        highest possible resolution found in both input and output as
+        reference.
+
+        """
+        new_config = super().to_baseline()
+        resolutions = self.input_resolutions + self.output_resolutions
+        if self.save_resolution is not None:
+            resolutions.append(self.save_resolution)
+
+        scale_factors = self.scale_to_highest(resolutions, self.resolution_unit)
+        num_input_resolutions = len(self.input_resolutions)
+        num_output_resolutions = len(self.output_resolutions)
+
+        end_idx = num_input_resolutions + num_output_resolutions
+        output_resolutions = [
+            {"units": "baseline", "resolution": v}
+            for v in scale_factors[num_input_resolutions:end_idx]
+        ]
+
+        save_resolution = None
+        if self.save_resolution is not None:
+            save_resolution = {"units": "baseline", "resolution": scale_factors[-1]}
+
+        return replace(
+            self,
+            input_resolutions=new_config.input_resolutions,
+            output_resolutions=output_resolutions,
+            save_resolution=save_resolution,
+        )
+
+
+class IOPatchPredictorConfig(ModelIOConfigABC):
+    """Contains patch predictor input and output information.
+
+    Args:
+        input_resolutions (list(dict)):
+            Resolution of each input head of model inference, must be in
+            the same order as `target model.forward()`.
+        patch_input_shape (:class:`numpy.ndarray`, list(int), tuple(int)):
+            Shape of the largest input in (height, width).
+        stride_shape (:class:`numpy.ndarray`, list(int), tuple(int)):
+            Stride in (x, y) direction for patch extraction.
+        output_resolutions (list(dict)):
+            Resolution of each output head from model inference, must be
+            in the same order as target model.infer_batch().
+
+    Attributes:
+        input_resolutions (list(dict)):
+            Resolution of each input head of model inference, must be in
+            the same order as `target model.forward()`.
+        patch_input_shape (:class:`numpy.ndarray`, list(int), tuple(int)):
+            Shape of the largest input in (height, width).
+        stride_shape (:class:`numpy.ndarray`, list(int), tuple(int)):
+            Stride in (x, y) direction for patch extraction.
+        output_resolutions (list(dict)):
+            Resolution of each output head from model inference, must be
+            in the same order as target model.infer_batch().
+        highest_input_resolution (dict):
+            Highest resolution to process the image based on input and
+            output resolutions. This helps to read the image at the optimal
+            resolution and improves performance.
+
```

Comment:
missing example

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/618/commits/07081308f0e0805cd569baac8a7b8628c586a46f